airbyte-cdk/python/airbyte_cdk/sources/declarative/async_job/job_orchestrator.py

```diff
@@ -222,14 +225,24 @@ def create_and_get_completed_partitions(self) -> Iterable[AsyncPartition]:
             Each partition is wrapped in an Optional, allowing for None values.
         """
         while True:
-            self._start_jobs()
+            try:
+                self._start_jobs()
+            except self._exceptions_to_break_on as e:
+                self._abort_all_running_jobs()
+                raise e
+            except Exception as e:
+                LOGGER.error(f"Failed to start the Job: {e}, traceback: {traceback.format_exc()}")
             if not self._running_partitions:
                 break
 
             self._update_jobs_status()
             yield from self._process_running_partitions_and_yield_completed_ones()
             self._wait_on_status_update()
 
+    def _abort_all_running_jobs(self):
+        [self._job_repository.abort(job) for job in self._get_running_jobs()]
```

Comment:
nit: assuming there were timeout jobs, should we abort them as well?

Reply:
hmm, i think that timeouted jobs can not be aborted?

Reply:
That's what source-salesforce does today: https://github.com/airbytehq/airbyte/blob/8d04d840ba4d62634c98b37fbe3ed053e3ab7fab/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L469-L473 

That being said, I think I've been mixing two concepts and it shows in the code:
* We decide a job is timed out: in this case, we can abort and it should be fine
* The API tells us that sometime is timed out (which we allow [here](https://github.com/airbytehq/airbyte/blob/8d04d840ba4d62634c98b37fbe3ed053e3ab7fab/airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml#L2388-L2391)). In that case, we probably can't abort and the user should flag this state as `failed` instead. This means that we should probably remove lines 2388 to 2391...

Does the reasoning above sounds fair to you?

Reply:
Yes, I see, that makes sense

Reply:
updated in https://github.com/airbytehq/airbyte/pull/45400/commits/7e6f33386f3f98e48922c7862593bda022580f82